src/librustc_resolve/lib.rs

```diff
@@ -1921,18 +1921,15 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                 -> ResolveResult<(Rc<Module>, LastPrivate)> {
         fn search_parent_externals(needle: Name, module: &Rc<Module>)
                                 -> Option<Rc<Module>> {
-            module.external_module_children.borrow()
-                                            .get(&needle).cloned()
-                                            .map(|_| module.clone())
```

Comment:
heh, am I reading this right? It was basically double-cloning the data? Bulk-refactors are a hell of a drug. :)